src/libxtp/dftengine/dftengine.cc

```diff
@@ -297,6 +297,25 @@ bool DFTEngine::Evaluate(Orbitals& orb) {
       XTP_LOG(Log::error, *_pLog)
           << TimeStamp() << " Final Single Point Energy "
           << std::setprecision(12) << totenergy << " Ha" << flush;
+      XTP_LOG(Log::error, *_pLog) << TimeStamp() << std::setprecision(12)
+                                  << " Final Local Exc contribution "
+                                  << e_vxc.energy() << " Ha" << flush;
+      if (_ScaHFX > 0) {
+        XTP_LOG(Log::error, *_pLog)
+            << TimeStamp() << std::setprecision(12)
+            << " Final Non Local Ex contribution " << exx << " Ha" << flush;
+      }
+
+      Mat_p_Energy EXXs = CalcEXXs(MOs.eigenvectors(), Dmat);
+      exx = -1.0 / 4.0 * EXXs.energy();
+      XTP_LOG(Log::error, *_pLog) << TimeStamp() << std::setprecision(12)
+                                  << " EXX energy " << exx << " Ha" << flush;
+
+      XTP_LOG(Log::error, *_pLog)
+          << TimeStamp() << std::setprecision(12) << " Final EXX Total energy "
+          << totenergy - e_vxc.energy() + (1.0 - _ScaHFX) * exx << " Ha"
+          << flush;
+
```

Comment:
how does this relate to the change and what is it supposed to do?

Reply:
This will give the user the opportunity to calculate the total energy of the system in the RPA approximation. For that you need to subtract the Exc contribution from the total DFT energy, add HF-like Ex unscaled. This yields the so called EXX energy (eXact eXchange) to which the RPA correlation energy needs to be added. See Eq.1 in https://aip.scitation.org/doi/10.1063/1.3442749 (they call EXX EHF)

E^RPA = E^XX + E^RPA_correlation

Reply:
so the RPA stuff only works with xtpdft? How do you want to add this to the RPA? Or is the user supposed to add the contributions up by hand to get E^RPA

Reply:
I do not really understand the workflow here.

Reply:
Yes, unless we want to make the EXX energy part of orbitals. I suppose one could get something out of orca's logfile but it is nasty if for example the DFT calculation was done without hybrid.

Reply:
hmm.. would it be possible to simply calculate this in the gw part instead of the DFT?

Reply:
Maybe. Vxc for sure, not immediately clear what about the exchange energy, since I think that is part of Sigma, and not returning the total energy.

Reply:
you can add another function to sigma.

Reply:
That's above my paygrade

Reply:
? I think you can calculate Exx from the Sigma_x function via:

<a href="https://www.codecogs.com/eqnedit.php?latex=E^{xx}=Tr(D*V^{xx})&space;=>&space;V^{xx}_{\alpha\beta}=\sum_{\gamma\delta}D_{\gamma\delta}(\alpha\gamma|\beta\delta)=>&space;D_{\alpha\beta}=\sum_i^{occ}&space;M_{\alpha&space;i}M_{i\beta}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?E^{xx}=Tr(D*V^{xx})&space;=>&space;V^{xx}_{\alpha\beta}=\sum_{\gamma\delta}D_{\gamma\delta}(\alpha\gamma|\beta\delta)=>&space;D_{\alpha\beta}=\sum_i^{occ}&space;M_{\alpha&space;i}M_{i\beta}" title="E^{xx}=Tr(D*V^{xx}) => V^{xx}_{\alpha\beta}=\sum_{\gamma\delta}D_{\gamma\delta}(\alpha\gamma|\beta\delta)=> D_{\alpha\beta}=\sum_i^{occ} M_{\alpha i}M_{i\beta}" /></a>

if you put that all in you should get a nice function for Exx in terms of (ij|kl),i.e. the 4center integrals in MO basis, which are used in the GW calculation

Reply:
Yes, in principle. I practice I am not so sure what happens if the GW corrections are not requested for all occupied states, for instance, and the the 4c in MO basis are not the same in Sigma_x as needed for E_x.

Reply:
Although this maybe on RPAlevels level, but I have no time to dig through that now.

Reply:
If you do not do a full range calculation you get the same problem for the correlation contribution as well.

Reply:
Not exactly the same. You may have an E_c calculated with a limited range set (which one might physically) but at least the E_x is the same as used in the hybrid DFT.

Reply:
I will merge this as is and put up another issue to extend the EXX total energy calculation to other codes by either having it in the qmpackage or in the Sigma.